packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -359,6 +385,55 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
             .map((fieldKey) => this.proxifyField(fieldKey, false))
             .values();
     }
+
+    public createField(fieldKey: FieldKey, newContent: ITreeCursor): EditableField | undefined {
+        assert(!this.has(fieldKey), "The field already exists.");
+        const fieldKind = this.lookupFieldKind(fieldKey);
+        assert(
+            fieldKind.multiplicity !== Multiplicity.Value,
+            "It is invalid to create fields of kind `value` as they should always exist.",
+        );
+        const path = this.cursor.getPath();
+        switch (fieldKind.multiplicity) {
+            case Multiplicity.Optional: {
+                if (this.context.setOptionalField(path, fieldKey, newContent, true))
+                    return this.proxifyField(fieldKey, false);
+            }
+            case Multiplicity.Sequence: {
+                if (this.context.insertNodes(path, fieldKey, 0, newContent))
+                    return this.proxifyField(fieldKey, false);
+            }
+            default:
+        }
+        return undefined;
+    }
+
+    public deleteField(fieldKey: FieldKey): void {
+        assert(this.has(fieldKey), "The field does not exist.");
```

Comment:
In my opinion, deleting an empty sequence or optional field shouldn't be an error. Doing so might even be meaningful when it comes to merges (if you want to ensure some field is empty, then do something else, you might want to include the delete edit in the op, even if the field was already empty).

Reply:
We can let the multiplicity validation handle forbidden fields below, which will error if you try and delete fields that are not allowed at all by the schema (we could also make that a non-error, but I think it's fine for that to error)

Reply:
> In my opinion, deleting an empty sequence or optional field shouldn't be an error. Doing so might even be meaningful when it comes to merges (if you want to ensure some field is empty, then do something else, you might want to include the delete edit in the op, even if the field was already empty).

It is not, as the proxy handler just let it go through if the field does not exist. This assert is only internal to avoid misuses, typos etc.

Reply:
I must've read more carefully. I have removed this and also the `length === 0` check for fields.